src/source/source_cache.js

```diff
@@ -241,14 +241,14 @@ class SourceCache extends Evented {
             tile.state = 'errored';
             if ((err: any).status !== 404) this._source.fire(new ErrorEvent(err, {tile}));
             // continue to try loading parent/children tiles if a tile doesn't exist (404)
-            else this.update(this.transform);
+            else this.update(this.transform, undefined, this._source.type === 'raster-dem');
```

Comment:
It is good to evaluate what happens when when raster dem source is not used for terrain, but only for hillshade. userForTerrain property gets evaluated to true here.

Otherwise, when used for terrain, we need to cache tile size for this source supplied (it is scaled up in code below) and if in consecutive update here we end up requesting tile cover with different tile size.

https://github.com/mapbox/mapbox-gl-js/blob/cd46ea1da423e74e56510bcd4c268041f000846f/src/terrain/terrain.js#L281